### ISSUE TITLE
Hide autocomplete when no results

### ISSUE THREAD

**Issue by malles** [uikit/uikit#1050]
When the items for the autocomplete are fetched and filtered by the user input and the user types something that is not in the list, the dropdown remains visible with no options in it.
![screenshot_54](https://cloud.githubusercontent.com/assets/5442402/6543911/fb4df790-c532-11e4-807c-9a94d3e227b7.png)
![screenshot_53](https://cloud.githubusercontent.com/assets/5442402/6543913/fe994ec2-c532-11e4-8eca-92196dce7936.png)

When I tried to hide it on `show.uk.autocomplete` I could not do it, since the data is not passed [in that trigger](https://github.com/uikit/uikit/blob/master/src/js/components/autocomplete.js#L314), so there is no (easy) way to tell if there are any items.
So I had to pass my own renderer in through the options that hid the dropdown when there are no items. In fact, no problem, great that the option exists.
I noticed though that the event `show.uk.autocomplete` is not triggered when you use your own renderer (unless you trigger it there yourself of course). That seems odd.

Basically two questions:
- Could the dropdown be hidden when [`data.length == 0`](https://github.com/uikit/uikit/blob/master/src/js/components/autocomplete.js#L309).
- Could the data be passed in the `show.uk.autocomplete` event so customization can be easier done in a triggered function.

More examples: http://codepen.io/malles/pen/qEyEZr?editors=101

**Comment by mojtabakaviani** [comment]
Fixed in pull request #981

**Comment by malles** [comment]
Yes, I saw you sneak that in there. That's not the proper position because it won't hide if `data` but `!data.length`

**Comment by philoest** [comment]
We have released the next major version, UIkit3. To start focusing our work on the most recent release, we decided to close all the issues related to UIkit2.

Thanks for your understanding,
Your UIkit Team
